client/ngapp/scripts/controllers/todo.js

```diff
@@ -56,9 +62,11 @@ angular.module('loopbackExampleFullStackApp')
   });
 
   $scope.addTodo = function () {
-    var todo = new Todo({title: $scope.newTodo});
-    todo.save();
-    $scope.newTodo = '';
+    Todo.create({title: $scope.newTodo})
+      .then(function() {
+        $scope.newTodo = '';
+        $scope.$apply();
+      });
```

Comment:
@ritch do you happen to remember why your original code was cleaning `$scope.newTodo` before the `save` request finished? Is it important to preserve that behaviour, or is the new solution proposed by @superkhau ok too?